api/dao/liststorage.py

```diff
@@ -182,3 +182,49 @@ def _get_el(self, _id, query_params):
         result = self.dbc.find_one(query, projection)
         if result and result.get(self.list_name):
             return result.get(self.list_name)[0]
+
+
+class AnalysesStorage(ListStorage):
+
+    def get_fileinfo(self, _id, analysis_id, filename = None):
+        _id = bson.ObjectId(_id)
+        query = [
+            {'$match': {'_id' : _id}},
+            {'$unwind': '$' + self.list_name},
+            {'$match': {self.list_name+ '._id' : analysis_id}},
+            {'$unwind': '$' + self.list_name + '.files'}
+        ]
```

Comment:
It was probably already discussed at some point, but we might want to consider moving analysis objects to their own container level to prevent mongo difficulties when dealing with a list inside a list. For example, it would be impossible to edit a specific field of one of these files in an update query, as using `$` more than once in mongo is not allowed. Something to think about for future improvements. I would add an extra layer to our already complex hierarchy, though.